xonsh/ptk_shell/completer.py

```diff
@@ -99,6 +101,35 @@ def get_completions(self, document, complete_event):
                 disp = comp[pre:].strip("'\"")
                 yield Completion(comp, -plen, display=disp)
 
+    def __find_begidx(self, line, endidx):
+        if len(line[0:endidx]) == 0:
+            return 0
+
+        wordsNonPosix = []
+        try:
+            wordsNonPosix = split(
+                line[0:endidx] + "_", posix=False
+            )  # ensure last word is empty when ends with space
+            wordsNonPosix[-1] = wordsNonPosix[-1][:-1]
+        except ValueError:
+            try:
+                wordsNonPosix = split(
+                    line[0:endidx] + '"', posix=False
+                )  # try with missing `"` quote
+                wordsNonPosix[-1] = wordsNonPosix[-1][:-1]
+            except ValueError:
+                wordsNonPosix = split(
+                    line[0:endidx] + "'", posix=False
+                )  # try with missing `'` quote
+                wordsNonPosix[-1] = wordsNonPosix[-1][:-1]
```

Comment:
A good rule here with try/catch is to only put the lines likely to raise the error inside of the `try`, then put the followup lines outside of the try/catch block

Reply:
Probably would have to add a pseudo character in the first try block for this as its meant to remove the temporary quotation character which is only added in the other blocks.